common/searchattribute/search_attirbute.go

```diff
@@ -78,35 +78,15 @@ func BuildTypeMap(validSearchAttributesFn dynamicconfig.MapPropertyFn) (map[stri
 	return result, nil
 }
 
-// GetType returns type of search attribute from type map.
-func GetType(name string, typeMap map[string]enumspb.IndexedValueType) (enumspb.IndexedValueType, error) {
-	if len(typeMap) == 0 {
-		return enumspb.INDEXED_VALUE_TYPE_UNSPECIFIED, ErrTypeMapIsEmpty
-	}
-
-	saType, isDefined := typeMap[name]
-	if !isDefined {
-		return enumspb.INDEXED_VALUE_TYPE_UNSPECIFIED, fmt.Errorf("%w: %s", ErrInvalidName, name)
-	}
-	return saType, nil
-}
-
 // ApplyTypeMap set type for all valid search attributes which don't have it.
 // It doesn't do any validation and just skip invalid or already set search attributes.
-func ApplyTypeMap(searchAttributes *commonpb.SearchAttributes, typeMap map[string]enumspb.IndexedValueType) {
-	if len(typeMap) == 0 {
-		return
-	}
-
+func ApplyTypeMap(searchAttributes *commonpb.SearchAttributes, typeMap NameTypeMap) {
 	for saName, saPayload := range searchAttributes.GetIndexedFields() {
 		_, metadataHasValueType := saPayload.Metadata[MetadataType]
 		if metadataHasValueType {
 			continue
 		}
-		valueType, isDefined := typeMap[saName]
-		if !isDefined {
-			continue
-		}
+		valueType, _ := typeMap.GetType(saName)
```

Comment:
`valueType, _ := typeMap.GetType(saName)`

what if the saName is not present?

Reply:
in case of error `typeMap.GetType` returns type as `enumspb.INDEXED_VALUE_TYPE_UNSPECIFIED` which is properly handled by next call to `setMetadataType`.  I can add something like:
```
if err != nil {
    continue
}
```
because if `saName` is not defined (which should never happened under normal circumstances), there is no way to set it type.

Reply:
let us try to decrease the number of implicit contracts, 
either add a comment explaining the behavior, or make the error case / edge case explicit

Reply:
Added the snippet above.